iceoryx_hoofs/source/posix_wrapper/file_lock.cpp

```diff
@@ -92,8 +92,8 @@ FileLock& FileLock::operator=(FileLock&& rhs) noexcept
 
         CreationPattern_t::operator=(std::move(rhs));
 
-        m_name = std::move(rhs.m_name);
-        m_fd = std::move(rhs.m_fd);
+        m_name = std::move(rhs.m_name); // NOLINT TODO(r7vme)
```

Comment:
@elfenpiff @elBoberido  All parts marked with `TODO(r7vme)` are related to usage of `CreationPattern_t::operator=(std::move(rhs));` and std::move afterwards, which means usage of moved objects (which is violation).

What `CreationPattern_t::operator=(std::move(rhs));` does? I don't understand completely this part

Reply:
@r7vme 
Hey this is absolutely correct and weird. The problem is the following:
- MemoryMap inherits from designpattern::creation
- creation has two members m_isInitialized and m_errorValue which have to be moved with every move operation
- creation has an explicit move ctor and move assignment operator which invalidates the source after a move operation - they should always be used

When we now implement a resource handling class which inherits from creation it has to call first the move ctor/assignment operator before moving the members to the new source otherwise the two members m_isInitialized and m_errorValue are either undefined or we set them manually and then the origin is not invalidated
We solve this by delegating the move operation first to the parent so that the members are moved correctly.